service/history/mutableStateBuilder.go

```diff
@@ -1125,6 +1125,10 @@ func (e *mutableStateBuilder) FailDecision(incrementAttempt bool) {
 	e.UpdateDecision(failDecisionInfo)
 }
 
+func (e *mutableStateBuilder) ClearDecisionAttempt() {
```

Comment:
Can we get rid of this API and have code in mutableStateBuilder to reset attempt when decision is failed because of Reset error?

Reply:
ok